client/src/components/settings/Username.js

```diff
@@ -77,7 +78,8 @@ class UsernameSettings extends Component {
       /* eslint-disable-next-line react/no-did-update-set-state */
       return this.setState({
         isFormPristine: username === formValue,
-        submitClicked: false
+        submitClicked: false,
+        isUserNew: prevUsername === ''
```

Comment:
I thought new users have `'fcc' + uuid()` as their username to start with.  If so, won't `isUserNew` always be false?

Reply:
The idea is that a new user won’t have a previous username so that’ll set the flags for the message. As I’ve described above, it works with developementuser on a local instance. Try and change the username on the code from this fork once.

Reply:
Sure, but developmentuser is a special dev only user.  As I understand it a real user gets their first `username` from here: https://github.com/freeCodeCamp/freeCodeCamp/blob/316d56ba11a95cb16982d9a86382f8d00ea5e099/api-server/common/models/user.js#L194

I'm no expert on the server-side stuff, though, so I could easily have misunderstood how this works.

Reply:
Hmm. I could try checking for the starting being fcc but that would flag users like fccnameone too. I guess @Bouncey can provide us some insight?

Reply:
Yep, there are people with usernames that start with fcc.  One workaround would be to check for usernames of the form
```js
'fcc1900a90a-86be-4c54-9846-d371972e9c48'
```
where each symbol after fcc is a hexadecimal.  Users that changed their names between uuids wouldn't see the warning, but I'm guessing that's rare.

Reply:
All users have a `username`, the initial user name is in the format of `fcc + uuid.v4()` (I have no idea why it is prefixed with fcc).

`prevUsername` will never be an empty string.

Reply:
I've given the RegEx idea a go. How do you guys think this looks?